delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/eventlog/ExpandedGlobalEventLog.scala

```diff
@@ -0,0 +1,74 @@
+package ch.epfl.bluebrain.nexus.delta.service.eventlog
+
+import akka.persistence.query.Offset
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.ExpandedJsonLd
+import ch.epfl.bluebrain.nexus.delta.sdk.eventlog.{EventExchangeCollection, GlobalEventLog}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.OrganizationRejection.OrganizationNotFound
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRejection.ProjectNotFound
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Event, Label, ResourceF}
+import ch.epfl.bluebrain.nexus.delta.sdk.{Organizations, Projects}
+import ch.epfl.bluebrain.nexus.sourcing.EventLog
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.{IO, Task}
+
+/**
+  * An implementation of [[GlobalEventLog]] for [[ExpandedJsonLd]]
+  */
+final class ExpandedGlobalEventLog private (
+    eventLog: EventLog[Envelope[Event]],
+    projects: Projects,
+    orgs: Organizations,
+    eventExchanges: EventExchangeCollection
+) extends GlobalEventLog[ResourceF[ExpandedJsonLd]] {
+
+  private val logger: Logger = Logger[ExpandedGlobalEventLog]
+
+  override def events(offset: Offset): Stream[Task, ResourceF[ExpandedJsonLd]] = exchange(
+    eventLog.eventsByTag(Event.eventTag, offset)
```

Comment:
We could add a TODO here to add an operation to just keep the last event for a given persistenceId (and go back to it later)

Reply:
I don't see the signature to that `EventExchange` more useful than the current. The JsonLdEncoder[S] part is misleading there, since in many cases you don't want to perform any encoding (it is already present in the `State`).